# HG changeset patch
# User Ken Brown <kbrown@cornell.edu>
# Date 1425524735 18000
#      Wed Mar 04 22:05:35 2015 -0500
# Node ID df85b49c7315d4c365f32455f4fdeee2893c48b4
# Parent  34be157e71d92a16c20bb601828cee54156ad2a5
* src/lispbibl.d: Use LINUX_NOEXEC_HEAPCODES on 32-bit Cygwin

The heap on Cygwin is now in high memory, so STANDARD_HEAPCODES will
no longer work.

diff -r 34be157e71d9 -r df85b49c7315 src/lispbibl.d
--- a/src/lispbibl.d	Wed Mar 04 21:54:40 2015 -0500
+++ b/src/lispbibl.d	Wed Mar 04 22:05:35 2015 -0500
@@ -2506,7 +2506,7 @@
      malloc results (and hence also of shared libraries) are randomized;
      only the code address is fixed around 0x1C000000 and the stack address
      is around 0xCF000000. In this case, we also use LINUX_NOEXEC_HEAPCODES. */
-    #if (defined(I80386) && defined(UNIX_LINUX)) || (defined(I80386) && defined(UNIX_OPENBSD) && defined(ADDRESS_RANGE_RANDOMIZED))
+    #if (defined(I80386) && defined(UNIX_LINUX)) || (defined(I80386) && defined(UNIX_OPENBSD) && defined(ADDRESS_RANGE_RANDOMIZED)) || (defined(I80386) && defined(UNIX_CYGWIN32))
       #define LINUX_NOEXEC_HEAPCODES
     #else
       #define STANDARD_HEAPCODES
